migrations/versions/8-recommendation_primary_key.py

```diff
@@ -0,0 +1,32 @@
+"""change primary key type from string to UUID
+
+Create Date: 2024-04-21 02:08:19.548067
+"""
+
+from collections.abc import Sequence
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "d99516f0bbcb"
+down_revision: str | None = "e3d06397960d"
+branch_labels: str | Sequence[str] | None = None
+depends_on: str | Sequence[str] | None = None
+
+
+def upgrade() -> None:
+    op.alter_column(
+        "recommendation",
+        "id",
+        type_=sa.types.Uuid(),
+        postgresql_using="id::uuid",
+    )
```

Comment:
To be able to use it with SQLite, could you wrap alters in a [batch migration bloc](https://alembic.sqlalchemy.org/en/latest/batch.html) ?

```suggestion
    with op.batch_alter_table("recommendation") as batch_op:
        batch_op.alter_column(
            "id",
            type_=sa.types.Uuid(),
            postgresql_using="id::uuid",
        )
```